src/main/java/ch/njol/skript/expressions/ExprNewBannerPattern.java

```diff
@@ -0,0 +1,85 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Color;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.banner.Pattern;
+import org.bukkit.block.banner.PatternType;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Banner Pattern")
+@Description("Create a new banner pattern.")
+@Examples({
+	"set {_pattern} to a new creeper banner pattern colored red",
+	"add {_pattern} to banner patterns of {_banneritem}",
+	"remove {_pattern} from banner patterns of {_banneritem}",
+	"set the 1st banner pattern of block at location(0,0,0) to {_pattern}",
+	"clear the 1st banner pattern of block at location(0,0,0)",
+	"",
+	"set {_pattern} to a red mojang banner pattern"
+})
+@Since("INSERT VERSION")
+public class ExprNewBannerPattern extends SimpleExpression<Pattern> {
+
+	static {
+		Skript.registerExpression(ExprNewBannerPattern.class, Pattern.class, ExpressionType.PATTERN_MATCHES_EVERYTHING,
+			"[a] %bannerpatterntype% colo[u]red %*color%",
```

Comment:
```suggestion
			"[a] %bannerpatterntype% colo[u]red %color%",
```
it'd be a bit silly to not allow dynamic colors for patterns at all
it's only required for the second one to help the parser